patroni/postgresql.py

```diff
@@ -1041,6 +1044,38 @@ def basebackup(self, conn_url, env):
 
         return ret
 
+    def pick_synchronous_standby(self, cluster):
+        current = cluster.sync.sync_standby if cluster.sync else None
+        members = {m.name: m for m in cluster.members}
+        for app_name, state, sync_state in self.query(
+                """SELECT application_name, state, sync_state
+                    FROM pg_stat_replication
+                    ORDER BY sync_state = 'sync' DESC, flush_location DESC"""):
+            member = members.get(app_name)
+            if not member:
+                continue
+            if member.tags.get('nosync', False):
+                continue
+            if state != 'streaming':
```

Comment:
I thought about adding some debug logging here so users gain some visibility into why nodes are or are not picked as sync standbys. That said, until that logging actually gets added this can be in the query.